core/object.cpp

```diff
@@ -1483,6 +1483,8 @@ void Object::_bind_methods() {
 		mi.name = "emit_signal";
 		mi.arguments.push_back(PropertyInfo(Variant::STRING, "signal"));
 
+		mi.return_val = PropertyInfo(Variant::NIL, "", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_NIL_IS_VARIANT);
```

Comment:
Is it only those methods? Or do all vararg methods need that? Do all other vararg methods have this?

Reply:
Apparently yes, all current varargs methods need that. So maybe it makes sense to move that code into bind_vararg_method? But I'm not sure if it makes sense to consider that all varargs methods return Variant - because "varargs" in general is about variable number of arguments, and not related to the return value.

Reply:
Actually, no. For example, GDScript's "new" returns Object, so I'll just add this line where it's needed.

Reply:
You don't need this. I'll change it in #10745 so it's detected automatically.

Reply:
I mean you don't need it for vararg methods. Virtual methods still need explicit types.

Reply:
Why does this PR need this? And are those the only places it's needed? It seems a bit fishy to me.

Apart from that the PR looks good, it's just those `return_val` things that grind my gears a bit.

Reply:
Like I said, it's not needed. I fixed it in #10745 for vararg methods to automatically have this. Virtual methods still need it, but I also fixed all the places where it was missing.

Reply:
Then @endragor should change that 😄

Reply:
@endragor That PR got merged, can you remove those lines and try again?